File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetscommercepayment.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CodatDataContractsDatasetsCommercePaymentMethodRef } from "./codatdatacontractsdatasetscommercepaymentmethodref";
import { CodatDataContractsDatasetsCommercePaymentStatusEnum } from "./codatdatacontractsdatasetscommercepaymentstatusenum";



export class CodatDataContractsDatasetsCommercePayment extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=amount" })
  amount?: number;

  @SpeakeasyMetadata({ data: "json, name=createdDate" })
  createdDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=currency" })
  currency?: string;

  @SpeakeasyMetadata({ data: "json, name=dueDate" })
  dueDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=modifiedDate" })
  modifiedDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=paymentMethodRef" })
  paymentMethodRef?: CodatDataContractsDatasetsCommercePaymentMethodRef;

  @SpeakeasyMetadata({ data: "json, name=paymentProvider" })
  paymentProvider?: string;

  @SpeakeasyMetadata({ data: "json, name=sourceModifiedDate" })
  sourceModifiedDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: CodatDataContractsDatasetsCommercePaymentStatusEnum;
}

File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetscommercepayment.go
package shared

import (
	"time"
)

type CodatDataContractsDatasetsCommercePayment struct {
	Amount             *float64                                             `json:"amount,omitempty"`
	CreatedDate        *time.Time                                           `json:"createdDate,omitempty"`
	Currency           *string                                              `json:"currency,omitempty"`
	DueDate            *time.Time                                           `json:"dueDate,omitempty"`
	ID                 *string                                              `json:"id,omitempty"`
	ModifiedDate       *time.Time                                           `json:"modifiedDate,omitempty"`
	PaymentMethodRef   *CodatDataContractsDatasetsCommercePaymentMethodRef  `json:"paymentMethodRef,omitempty"`
	PaymentProvider    *string                                              `json:"paymentProvider,omitempty"`
	SourceModifiedDate *time.Time                                           `json:"sourceModifiedDate,omitempty"`
	Status             *CodatDataContractsDatasetsCommercePaymentStatusEnum `json:"status,omitempty"`
}
